### DISCUSSION TITLE
where the Blade-type control interface can be found

### DISCUSSION THREAD

**Discussion by Tomedssa** [NatLabRockies/ROSCO#277] [Announcements]
Hi,
 I just started to contact ROSCO, some questions may not be professional, please don't mind. The article "A Reference Open Source Controller for Stationary and floating offshore wind Turbines" mentions that OpenFAST uses a Blade-type control interface to communicate with ROSCO. But I'm not sure where the Blade-type control interface can be found. Can you help me find it?
Thank you very much for your time in advance.
Tomedssa

**Comment by dzalkind** [comment]
The bladed interface is the set of variables (called the `avrSWAP`) passed between OpenFAST and ROSCO.  It can be found [here in ROSCO](https://github.com/NREL/ROSCO/blob/e3b7db779ad9e7bea5dea692e92f52b10116cf02/ROSCO/src/ReadSetParameters.f90#L39) and [here in OpenFAST](https://github.com/OpenFAST/openfast/blob/6a6511adc44e0b35676e27e1a27caced1ae8c152/modules/servodyn/src/BladedInterface.f90#L954).
